internal/systemd/systemd_test.go

```diff
@@ -5,80 +5,253 @@ import (
 		t.Skip("DBUS_SESSION_BUS_ADDRESS undefined")
 	}
 
 	tests := []struct {
-		description string
-		input       struct {
-			unitFile string
-			now      bool
-		}
-		want      string
-		wantError error
+		name           string
+		connectionType ConnectionType
+		expectError    bool
 	}{
 		{
-			description: "activate simple.service",
-			input: struct {
-				unitFile string
-				now      bool
-			}{
-				unitFile: "testdata/simple.service",
-				now:      true,
-			},
-			want: "active",
+			name:           "user_connection",
+			connectionType: ConnectionTypeUser,
+			expectError:    false,
+		},
+		{
+			name:           "system_connection",
+			connectionType: ConnectionTypeSystem,
+			expectError:    false,
 		},
 	}
 
-	for _, test := range tests {
-		t.Run(test.description, func(t *testing.T) {
-			conn, err := NewConnectionContext(context.Background(), ConnectionTypeUser)
-			if err != nil {
-				t.Fatal(err)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			conn, err := NewConnectionContext(context.Background(), tt.connectionType)
+
+			if tt.expectError {
+				if err == nil {
+					t.Error("expected error but got none")
+				}
+				return
 			}
 
-			abs, err := filepath.Abs(test.input.unitFile)
-			if err != nil {
-				t.Fatal(err)
+			if tt.connectionType == ConnectionTypeSystem && err != nil {
+				t.Skipf("System connection failed: %v", err)
+				return
 			}
 
-			// Link the testdata unit files into the search paths so they can be
-			// referenced by name. This simulates the context in which this
-			// package is more likely to be run.
-			_, err = conn.conn.LinkUnitFilesContext(conn.ctx, []string{abs}, true, false)
 			if err != nil {
-				t.Fatal(err)
+				t.Fatalf("unexpected error: %v", err)
 			}
 
-			err = conn.EnableUnit(filepath.Base(test.input.unitFile), test.input.now, true)
-			defer func(t *testing.T) {
-				if err := conn.DisableUnit(filepath.Base(test.input.unitFile), test.input.now, true); err != nil {
-					t.Fatal(err)
-				}
-			}(t)
-			if test.wantError != nil {
-				if !cmp.Equal(err, test.wantError, cmpopts.EquateErrors()) {
-					t.Errorf("%#v != %#v", err, test.wantError)
-				}
-			} else {
-				if err != nil {
-					t.Fatal(err)
-				}
-				got, err := conn.GetUnitState(filepath.Base(test.input.unitFile))
-				if err != nil {
-					t.Fatal(err)
-				}
-				if !cmp.Equal(got, test.want) {
-					t.Errorf("%v", cmp.Diff(got, test.want))
-				}
+			if conn == nil {
+				t.Error("expected non-nil connection")
+				return
 			}
+
+			conn.Close()
 		})
 	}
 }
+
+func TestReload(t *testing.T) {
+	if _, has := os.LookupEnv("DBUS_SESSION_BUS_ADDRESS"); !has {
+		t.Skip("DBUS_SESSION_BUS_ADDRESS undefined")
+	}
+
+	conn, err := NewConnectionContext(context.Background(), ConnectionTypeUser)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	err = conn.Reload()
+	if err != nil {
+		t.Logf("Reload failed: %v", err)
+	}
+}
+
+func TestUnitOperationsValid(t *testing.T) {
```

Comment:
Please add here some comment saying that all functions calls should not return error in normal situation.